testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/mockprovider/skipquerydetach/SFSB1.java

```diff
@@ -0,0 +1,116 @@
+        if(TestEntityManager.getInvocations().contains("close")) {
+            return "invalid state, entity manager has already been previously closed";
+        }
+        if(TestEntityManager.getInvocations().contains("clear")) {
+            return "invalid state, entity manager has already been previously cleared";
+        }
+        Query query =  em.createQuery("mock query");
+        if(TestEntityManager.getInvocations().contains("close")) {
+            return "invalid state, entity manager was closed before query was executed, " +
+                    "which means that 'wildfly.jpa.skipquerydetach=true' couldn't work, as the " +
+                    "persistence context was closed too early";
+        }
+        if(TestEntityManager.getInvocations().contains("clear")) {
+            return "invalid state, entity manager was cleared (detached) before query was executed, " +
+                    "which means that 'wildfly.jpa.skipquerydetach=true' didn't work";
+        }
+        query.getSingleResult();
+        if(TestEntityManager.getInvocations().contains("close")) {
+            return "invalid state, entity manager was closed before SFSB1 bean call completed, " +
+                    "which means that 'wildfly.jpa.skipquerydetach=true' couldn't work, as the " +
+                    "persistence context was closed too early";
+        }
+        if(TestEntityManager.getInvocations().contains("clear")) {
+            return "invalid state, entity manager was cleared (detached) before SFSB1 bean call completed, " +
+                    "which means that 'wildfly.jpa.skipquerydetach=true' didn't work";
+        }
+        return null;  // success
+    }
+
+    /**
+     * verify that entitymanager.clear is called after executing returned query.
+     *
+     * Return null for success, otherwise for test failure return fail message.
+     */
+    public String queryWithSkipQueryDetachDisabled() {
+        if(TestEntityManager.getInvocations().contains("close")) {
+            return "invalid state, entity manager has already been previously closed";
+        }
+        if(TestEntityManager.getInvocations().contains("clear")) {
+            return "invalid state, entity manager has already been previously cleared";
+        }
+        Query query =  emQueryResultsAreDetached.createQuery("mock query");
+        if(TestEntityManager.getInvocations().contains("clear")) {
+            return "invalid state, entity manager was cleared (detached) after query was created but before executed," +
+                    "which means that 'wildfly.jpa.skipquerydetach=false' (default) behaviour didn't work";
+        }
+        query.getSingleResult();
+        if(TestEntityManager.getInvocations().contains("close")) {
+            return "invalid state, entity manager was closed before SFSB1 bean call completed, " +
+                    "which means that 'wildfly.jpa.skipquerydetach=true' couldn't work, as the " +
```

Comment:
If I understand this correctly, we are testing the case with `wildfly.jpa.skipquerydetach=false`, aren't we? Should it be `false` in that case?

Reply:
oh, right, good catch, line 106 should be updated.  Thanks!